lib/cloudregister/registerutils.py

```diff
@@ -337,6 +337,50 @@ def get_current_smt():
     return smt
 
 
+# ----------------------------------------------------------------------------
+def get_instance_data(config):
+    """Run the configured instance data collection command and return
+       the result or none."""
+    instance_data = None
+    if (
+            config.has_section('instance') and
+            config.has_option('instance', 'dataProvider')
+    ):
+        instance_data_cmd = config.get('instance', 'dataProvider')
+        cmd = instance_data_cmd.split()[0]
+        if cmd != 'none':
+            if cmd[0] != '/':
+                try:
+                    p = subprocess.Popen(
+                        ['which %s' % cmd],
+                        stdout=subprocess.PIPE,
+                        stderr=subprocess.PIPE,
+                        close_fds=True
+                    )
+                except:
+                    errMsg = 'Could not find configured dataProvider: %s' % cmd
+                    logging.error(errMsg)
+            if os.access(cmd, os.X_OK):
+                try:
+                    p = subprocess.Popen(
+                        instance_data_cmd.split(),
```

Comment:
Suggest using shlex.split which will respect quotes and generally be less bug prone when splitting shell commands.